### PR TITLE
user.py: getting user with existing site group is now possible in get…

### PR DESCRIPTION
**Pull request by ProfessionalCaddie**
PLEASE ADVISE:
As you can see in my changes, I had to update all poetry dependencies in order to run tests on my machine. I know that is not what we may want but I am new to git so please give me some grace and clarity. Also after this change I WAS NOT able to use commands from the makefile. There might still be a good way to do this but I haven't learned the intricacies. 

For the reason above I didn't check the coding guidelines box.

# Pull Request

## Description
Logic has been added to the test_get_user_by_email function in user.py to be able to add a new user even if a site group already exists for their email

Fixes #

## How Has This Been Tested?

I added a test that creates a site group with an email already attached and then creates a user with the same email as the site group. This should prove the case because if another site group was created along side the new user, there would be 2 site groups and I assert that there is only one.

- [ ] Yes

_If your changes affect data processing, have you plotted any changes? i.e. have you done a quick sanity check?_

- [ ] Yes

## Checklist:

- [ ] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ x] I have added tests that prove my fix is effective or that my feature works
- [x ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by peterdudfield** [comment]
No problem, Thanks @ProfessionalCaddie for doing all this. 

do you mind not commiting any changes to the pyproject.toml files, or the poetry lock files. This should help the CI on our side

File: tests/read/test_get_user_by_email.py [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
 
 
 class TestGetUserByEmail:
     """Test for get_user_by_email function"""
-
+    def test_get_user_by_email_no_users_site_group_exists(self,db_session):
+        site_group = create_site_group(db_session=db_session,site_group_name="site_group_for_test@test.com")
+        user = get_user_by_email(session=db_session, email="test@test.com")
+        assert len(db_session.query(UserSQL).all()) == 1
+        assert len(db_session.query(SiteGroupSQL).all()) == 1
```
Comment (peterdudfield) [inline tests/read/test_get_user_by_email.py:12]: good test

**Comment by ProfessionalCaddie** [comment]
> No problem, Thanks @ProfessionalCaddie for doing all this.
> 
> do you mind not commiting any changes to the pyproject.toml files, or the poetry lock files. This should help the CI on our side

Hi @peterdudfield, I change pyproject.toml and poetry.lock to be the same as they are in your version of the project. If there is an easier or more preferred way to do this let me know.

**Comment by peterdudfield** [comment]
Ill merge it to `dev` incase I need to do any little adjustments

**Comment by peterdudfield** [comment]
Thanks @ProfessionalCaddie, ive got this all merged in now

@all-contributors please add @ProfessionalCaddie  for code

**Pull request merged**